### PR TITLE
 Adding unit test cases for registry resource

### PR DESCRIPTION
**Pull request by cnapagoda**
## Purpose
Adding Unit test cases registry resource component.

## Goals
Improve Unit test coverage of registry resource component.

### CODE REVIEW

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/README.txt [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#
+# Copyright 2009 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/README.txt]: Change the year of License header to 2007

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/README [hunk 2]
```diff
@@ -0,0 +1,116 @@
+CARBON_HOME/repository/conf directory contains all the Carbon configuration files.
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/README]: License header is missing. Please check

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/axis2/axis2.xml [hunk 3]
```diff
@@ -0,0 +1,701 @@
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/axis2/axis2.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/axis2/axis2_client.xml [hunk 4]
```diff
@@ -0,0 +1,299 @@
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/axis2/axis2_client.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/axis2/tenant-axis2.xml [hunk 5]
```diff
@@ -0,0 +1,274 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+
+<!--
+  ~ Copyright WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/axis2/tenant-axis2.xml]: License header year is missing.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/carbon.xml [hunk 6]
```diff
@@ -0,0 +1,609 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/carbon.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/datasources/master-datasources.xml [hunk 7]
```diff
@@ -0,0 +1,84 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+
+<!--
+  ~ Copyright WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/datasources/master-datasources.xml]: License header year is missing.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/email/email-admin-config.xml [hunk 8]
```diff
@@ -0,0 +1,42 @@
+<!--
+ ~ Copyright (c) 2005-2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/email/email-admin-config.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/bundle-config/README.txt [hunk 9]
```diff
@@ -0,0 +1,29 @@
+# ----------------------------------------------------------------------------
+#  Copyright 2015 WSO2, Inc. http://www.wso2.org
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/bundle-config/README.txt]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/cache.xml [hunk 10]
```diff
@@ -0,0 +1,42 @@
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/cache.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/config-validation.xml [hunk 11]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+<!--
+~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/config-validation.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/jgroups-s3_ping-aws.xml [hunk 12]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/jgroups-s3_ping-aws.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/jmx.xml [hunk 13]
```diff
@@ -0,0 +1,32 @@
+<!--
+~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/jmx.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/launch.ini [hunk 14]
```diff
@@ -0,0 +1,265 @@
+# ----------------------------------------------------------------------------
+#  Copyright 2015 WSO2, Inc. http://www.wso2.org
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/launch.ini]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/logging-bridge.properties [hunk 15]
```diff
@@ -0,0 +1,82 @@
+# ----------------------------------------------------------------------------
+#  Copyright 2015 WSO2, Inc. http://www.wso2.org
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/logging-bridge.properties]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/mime.mappings [hunk 16]
```diff
@@ -0,0 +1,27 @@
+#
+# Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/mime.mappings]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/mime.types [hunk 17]
```diff
@@ -0,0 +1,734 @@
+#
+# Copyright 2005-2009 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/mime.types]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/osgi-debug.options [hunk 18]
```diff
@@ -0,0 +1,112 @@
+# ----------------------------------------------------------------------------
+#  Copyright 2015 WSO2, Inc. http://www.wso2.org
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/etc/osgi-debug.options]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/log4j.properties [hunk 19]
```diff
@@ -0,0 +1,166 @@
+#
+# Copyright 2009 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/log4j.properties]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/registry.xml [hunk 20]
```diff
@@ -0,0 +1,100 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/registry.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/security/authenticators.xml [hunk 21]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/security/authenticators.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/security/cipher-text.properties [hunk 22]
```diff
@@ -0,0 +1,45 @@
+# ----------------------------------------------------------------------------
+#  Copyright 2015 WSO2, Inc. http://www.wso2.org
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/security/cipher-text.properties]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/security/cipher-tool.properties [hunk 23]
```diff
@@ -0,0 +1,51 @@
+# ----------------------------------------------------------------------------
+#  Copyright 2015 WSO2, Inc. http://www.wso2.org
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/security/cipher-tool.properties]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/security/secret-conf.properties [hunk 24]
```diff
@@ -0,0 +1,54 @@
+# ----------------------------------------------------------------------------
+#  Copyright 2015 WSO2, Inc. http://www.wso2.org
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/security/secret-conf.properties]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/tomcat/carbon/META-INF/context.xml [hunk 25]
```diff
@@ -0,0 +1,31 @@
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/tomcat/carbon/META-INF/context.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/tomcat/carbon/WEB-INF/web.xml [hunk 26]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ ~ Copyright (c) 2005-2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/tomcat/carbon/WEB-INF/web.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/tomcat/web.xml [hunk 27]
```diff
@@ -0,0 +1,1220 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/conf/tomcat/web.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/deployment/README.txt [hunk 28]
```diff
@@ -0,0 +1,28 @@
+#
+# Copyright 2009 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/deployment/README.txt]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/deployment/server/axis2modules/modules.list [hunk 29]
```diff
@@ -0,0 +1,17 @@
+#
+# Copyright 2009 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/deployment/server/axis2modules/modules.list]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/deployment/server/axis2services/services.list [hunk 30]
```diff
@@ -0,0 +1,16 @@
+#
+# Copyright 2009 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/deployment/server/axis2services/services.list]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/logs/README.txt [hunk 31]
```diff
@@ -0,0 +1,17 @@
+#
+# Copyright 2009 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/logs/README.txt]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/resources/README.txt [hunk 32]
```diff
@@ -0,0 +1,20 @@
+#
+# Copyright 2009 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/carbon-home/repository/resources/README.txt]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/user-test/jdbc-readonly-test.xml [hunk 33]
```diff
@@ -0,0 +1,63 @@
+<!--
+ ~ Copyright (c) 2005-2011, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/user-test/jdbc-readonly-test.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/user-test/test-log4j.xml [hunk 34]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+~ Copyright (c) 2005-2011, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/user-test/test-log4j.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/user-test/user-mgt-clear.xml [hunk 35]
```diff
@@ -0,0 +1,131 @@
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/user-test/user-mgt-clear.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/user-test/user-mgt-multi-tenant.xml [hunk 36]
```diff
@@ -0,0 +1,74 @@
+<!--
+  ~ Copyright 2005-2011 WSO2, Inc. (http://wso2.com)
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/user-test/user-mgt-multi-tenant.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/user-test/user-mgt-registry-test.xml [hunk 37]
```diff
@@ -0,0 +1,100 @@
+<!--
+ ~ Copyright (c) 2005-2011, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/user-test/user-mgt-registry-test.xml]: Need to change to 2017 in license header.

File: components/registry/org.wso2.carbon.registry.resource/src/test/resources/user-test/user-mgt-test.xml [hunk 38]
```diff
@@ -0,0 +1,104 @@
+<!--
+ ~ Copyright (c) 2005-2011, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (daneshk) [inline components/registry/org.wso2.carbon.registry.resource/src/test/resources/user-test/user-mgt-test.xml]: Need to change to 2017 in license header.

**Comment by cnapagoda** [comment]
@daneshk Update the PR

**Pull request merged**